cdap-ui/app/cdap/components/DAG/Nodes/AlertPublisherNode/index.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import withStyles from '@material-ui/core/styles/withStyles';
+import { IAbstractNodeProps, AbstractNode } from 'components/DAG/Nodes/AbstractNode';
+import { genericNodeStyles } from 'components/DAG/Nodes/utilities';
+
+const styles = genericNodeStyles({
+  border: `1px solid #ffba01`,
+  '&.drag-hover': {
+    backgroundColor: 'rgba(255, 186, 1, 0.1)',
+  },
+});
+interface IAlertPublisherProps extends IAbstractNodeProps<typeof styles> {}
+class AlertPublisherComponent extends AbstractNode<IAlertPublisherProps> {
+  public type = 'alertpublisher';
+
+  public checkForValidIncomingConnection = (connObj) => {
+    if (connObj.connection.target.getAttribute('data-node-type') !== 'alertpublisher') {
```

Comment:
i'm not sure i understand this condition, why are we checking the target?

Reply:
- Every node will have `checkForValidIncomingConnection` condition to determine if the current connection is valid or not.
- We register this callback from each node in the `DAGRenderer` [init node function ](https://github.com/cdapio/cdap/blob/feature/ui-CDAP-14284-2/cdap-ui/app/cdap/components/DAG/DAGRenderer.tsx#L248-L250).
- When a `connection` event is fired when user connects two nodes we call all the callbacks from the nodes. 
- It is basically a [reduce on all the callbacks](https://github.com/cdapio/cdap/blob/feature/ui-CDAP-14284-2/cdap-ui/app/cdap/components/DAG/DAGRenderer.tsx#L253-L255) and the eventual reduce value should be `true` for the connection to be drawn. It it returns false then one of the nodes involved in the connection rejects this connection so we don't render the connection.

The advantage of this is having such custom logic specific to node within the node.

Hope I made sense.

Now that I think about it I think I can add this function as a static function and have it registered as not per instance of the node. Will add that change.